zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -1282,8 +1305,132 @@ components:
+      type: object
+      properties:
+        filter:
+          $ref: "#/components/schemas/UserTaskFilterRequest"
+    UserTaskSearchQueryResponse:
+      allOf:
+        - $ref: "#/components/schemas/SearchQueryResponse"
+      description: User task search query response.
+      type: object
+      properties:
+        items:
+          type: array
+          items:
+            $ref: "#/components/schemas/UserTaskItem"
+    UserTaskFilterRequest:
+      description: User task filter request.
+      type: object
+      properties:
+        key:
+          type: integer
+          format: int64
+        state:
+          type: string
+        assignee:
+          type: string
+        elementId:
+          type: string
+        candidateGroup:
+          type: string
+        candidateUser:
+          type: string
+        processDefinitionKey:
+          type: integer
+          format: int64
+        processInstanceKey:
+          type: integer
+          format: int64
+        creationDate:
+          $ref: "#/components/schemas/DateFilter"
+        completionDate:
+          $ref: "#/components/schemas/DateFilter"
+        followUpDate:
+          $ref: "#/components/schemas/DateFilter"
+        dueDate:
+          $ref: "#/components/schemas/DateFilter"
+        tenantIds:
+          type: string
+        bpmnProcessDefinitionId:
+          type: string
+    UserTaskItem:
+      type: object
+      properties:
+        key:
+          type: integer
+          format: int64
+        state:
+          type: string
+        assignee:
+          type: string
+        elementId:
+          type: string
+        elementInstanceKey:
+          type: integer
+          format: int64
+        candidateGroup:
+          type: array
+          items:
+            type: string
+        candidateUser:
+          type: array
+          items:
+            type: string
+        bpmnProcessId:
+          type: string
+        processDefinitionKey:
+          type: integer
+          format: int64
+        processInstanceKey:
+          type: integer
+          format: int64
+        formKey:
+          type: integer
+          format: int64
+        creationDate:
+          type: string
+          format: date-time
+        completionDate:
+          type: string
+          format: date-time
+        followUpDate:
+          type: string
+          format: date-time
+        dueDate:
+          type: string
+          format: date-time
+        tenantIds:
+          type: string
+        externalFormReference:
+          type: string
+        processDefinitionVersion:
+          type: integer
+          format: int32
+        customHeaders:
+          type: object
+          additionalProperties:
+            type: string
+    DateFilter:
+      type: object
+      properties:
+        from:
+          type: string
+          format: date-time
+        to:
+          type: string
+          format: date-time
```

Comment:
💭 I have been thinking about this for a while. Introducing this now creates certain expectations/standards that other query endpoints should follow so we're consistent. We'll get search options for date attributes with the advanced search topic (lower than, greater than, ...). How about we don't support date attribute filtering yet and wait for advanced search to enable this?

**Why?**: If we add `from` and `to` now, it's part of the API. We cannot simply remove this with 8.7 again in favor of the advanced search options (that will bring the same functionality but in a standardized way, as far as I can see) as this would be a breaking change in the API. Thus, I'd like to be a bit more conservative here with introducing constructs that we have also never talked about yet for the C8 REST API, to be honest. This construct has never been presented in our endpoint overview or anything. 

Let me know what you think. I assume this is something we had in the Tasklist component (V1) REST API?

Reply:
@houssain-barouni, happy for your thoughts/input here as well so you've seen this discussion.

Reply:
If we agree that we don't want to add date attribute filters yet, let's remove the code related to this from this PR, to be honest. Removing such a chunk again later from multiple places (YAML, different classes) is rather cumbersome and chances are high someones else starts using this as well.